Detect an unusable Home directory

s0600204 opened the pull request:
When Open/LibreOffice runs, it looks for a folder containing user-specific configuration in the invoking user's home directory. If it cannot find it, Open/LibreOffice attempts to create it.

For users running `unoconv` manually, this isn't a problem. 

For users attempting to run this on a webserver, the invoking user is the http daemon (e.g. `apache`, `nginx`, `lighttpd`), and the home folder is typically the webserver's root directory (e.g. `/srv/http`, `/var/www`).

Unfortunately, there appear to be some scenarios where this folder is read-only, even for the daemon (see #87, #449). This leads to errors, as Open/LibreOffice is unable to create its configuration folder.

(It is assumed that if the configuration folder already exists, it must be usable, else how would it have been created?)

This PR solves the issue by detecting this, emitting a warning message, and using a temporary folder as a temporary "Home".

----

An earlier version of this PR also added invocation arguments to permit a web-admin to specify an alternate location to be treated as "home". This functionality has been removed as it is better to set the `HOME` environment value before invoking `unoconv`.

pataquets commented:
+1. This could save a lot of pain and is an easy solution.
Not to mention that the simple presence of the command line option alone could save lots of hair-pulling to people by hinting possible causes to obscure LO exit codes.

s0600204 commented:
The problem this pull request was to resolve may or may not have been fixed in either d9810b7 or e4f65ed or both. Either way there are now merge conflicts, so I'm closing this request.

If, after testing, the problem still remains, I'll report back on #87.

dagwieers commented:
I prefer to keep it open until we have decided how to implement it in #87, if you don't mind.

s0600204 commented:
That's absolutely fine, it's your project.

s0600204 commented:
I wish to point out that (for me) the changes within this PR are no longer necessary - it appears that at some point over the past five years, dealing with an unset/unreadable/unwritable `HOME` has been resolved upstream within the LibreOffice code base.

I don't know whether the same is true for OpenOffice, but I feel that it shouldn't be that hard for folks to set the `HOME` environment value before invoking `unoconv`?

Thus, I'd like to propose this PR be closed (as "resolved upstream" or something).

Alternately, I can update this to resolve the merge-conflict. (Or close and reopen as a new PR if that's cleaner/preferable.)

s0600204 committed:
Detect an unusable Home directory

When Open/LibreOffice runs, it looks for a folder containing user-specific
configuration in the invoking user's home directory. If it cannot find it,
Open/LibreOffice attempts to create it.

For users running `unoconv` manually, this isn't a problem.

For users attempting to run this on a webserver, the invoking user is the http
daemon (e.g. `apache`, `nginx`, `lighttpd`), and the home folder is typically
the webserver's root directory (e.g. `/srv/http`, `/var/www`).

Unfortunately, there appear to be some scenarios where this folder is read-only,
even for the daemon (see #87, #449). This leads to errors, as Open/LibreOffice
is unable to create its configuration folder.

(It is assumed that if the configuration folder already exists, it must be
usable, else how would it have been created?)

This commit solves the issue by detecting this, emitting a warning message, and
using a temporary folder as a temporary "Home".

s0600204 commented:
Full rewrite to resolve merge conflict, and reduce the PR down to the minimal(ish) change required to achieve original intent.
